src/tlog/format.ts

```diff
@@ -111,6 +117,28 @@ function toProposedIntotoV002Entry(
   };
 }
 
+// Calculates the hash of a DSSE envelope for inclusion in a Rekor entry.
+// There is no standard way to do this, so the scheme we're using as as
+// follows:
+//  * payload is base64 encoded
+//  * signature is base64 encoded (only the first signature is used)
+//  * keyid is included ONLY if it is NOT an empty string
+//  * The resulting JSON is canonicalized and hashed to a hex string
```

Comment:
Yikes!

Reply:
Is it worth writing some kind of tests around this behaviour to make sure we don't deviate from it in future?

Reply:
Do we have some tests in src/tlog/verify.test.ts that create different kinds of envelopes with and without a keyid? Maybe with multiple sigs?